drivers/fxos8700/fxos8700.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) UC Berkeley
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @file
+ * @brief       Driver for the FXOS8700 3-axis accelerometer/magnetometer
+ *
+ * @author      Michael Andersen <m.andersen@cs.berkeley.edu>
+ * @author      Hyung-Sin Kim <hs.kim@cs.berkeleyedu>
+ */
+
+
+#include "periph/i2c.h"
+#include "xtimer.h"
+#include "fxos8700.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#define I2C_SPEED               I2C_SPEED_FAST
+
+#ifndef FXOS8700_RENEW_INTERVAL
+#define FXOS8700_RENEW_INTERVAL 1000000ul
+#endif
+
+#define ACCEL_ONLY_MODE         0x00
+#define MAG_ONLY_MODE           0x01
+#define HYBRID_MODE             0x03
+
+static fxos8700_measurement_t acc_cached, mag_cached;
+static uint32_t last_read_time;
+
+static int fxos8700_read_regs(fxos8700_t* dev, uint8_t reg, uint8_t* data, size_t len)
+{
+    i2c_acquire(dev->p.i2c);
+    if(i2c_read_regs(dev->p.i2c, dev->p.addr, reg, (char*) data, len) <= 0) {
+        DEBUG("[fxos8700] Can't read register 0x%x\n", reg);
+        i2c_release(dev->p.i2c);
+        return FXOS8700_BUSERR;
+    }
+    i2c_release(dev->p.i2c);
+
+    return FXOS8700_OK;
+}
+
+static int fxos8700_write_regs(fxos8700_t* dev, uint8_t reg, uint8_t* data, size_t len)
+{
+    i2c_acquire(dev->p.i2c);
+    if(i2c_write_regs(dev->p.i2c, dev->p.addr, reg, (char*) data, len) <= 0) {
+        DEBUG("[fxos8700] Can't write to register 0x%x\n", reg);
+        i2c_release(dev->p.i2c);
+        return FXOS8700_BUSERR;
+    }
+    i2c_release(dev->p.i2c);
+
+    return FXOS8700_OK;
+}
+
+int fxos8700_init(fxos8700_t* dev, const fxos8700_params_t *params)
+{
+    dev->p.i2c = params->i2c;
+    int rv;
+    uint8_t config;
+
+    if ((params->addr < 0x1C) || (params->addr > 0x1F)) {
+        DEBUG("[fxos8700] Invalid address\n");
+        return FXOS8700_ADDRERR;
+    }
+    dev->p.addr = params->addr;
+
+    i2c_acquire(dev->p.i2c);
+    if(i2c_init_master(dev->p.i2c, I2C_SPEED) != 0) {
+        DEBUG("[fxos8700] Can't initialize I2C master\n");
+        i2c_release(dev->p.i2c);
+        return FXOS8700_NOBUS;
+    }
+
+    rv = i2c_read_regs(dev->p.i2c, dev->p.addr, 0x0D, &config, 1);
+    if (rv != 1) {
+        i2c_release(dev->p.i2c);
+        DEBUG("[fxos8700] Could not read WHOAMI (%d)\n", rv);
+        return FXOS8700_NOBUS;
+    }
+    dev->whoami = config;
```

Comment:
Is there any use case for saving this value? It seems like it is never used by the implementation.